src/main/scala/mesosphere/marathon/core/task/termination/TaskKillService.scala

```diff
@@ -0,0 +1,46 @@
+package mesosphere.marathon.core.task.termination
+
+import mesosphere.marathon.core.task.Task
+
+import scala.concurrent.Future
+
+/**
+  * A service that handles killing tasks. This will take care about extra logic for lost tasks,
+  * apply a retry strategy and throttle kill requests to Mesos.
+  */
+trait TaskKillService {
+  // TODO: should we favor distinct vs overloaded functions? killTask, killTaskById, killTasks, killUnknownTask?
```

Comment:
I prefer the distinct option. It makes the code easier to read at the call sites.